R/bookdown_to_leanpub.R

```diff
@@ -434,13 +434,36 @@ make_embed_markdown <- function(url,
 get_chapters <- function(bookdown_index = file.path("docs", "index.html"),
                          base_url = NULL) {
   # Read in html
-  index_html <- suppressWarnings(try(xml2::read_html(paste(bookdown_index, collapse = "\n"))))
+  index_html <- suppressWarnings(try(xml2::read_html(bookdown_index)))
 
-  # Extract chapter nodes
+  # Extract chapter nodes the Rmd way
   nodes <- rvest::html_nodes(index_html, xpath = paste0("//", 'li[@class="chapter"]'))
 
-  if (length(nodes) > 0) {
-    # Format into a data.frame
+  # If the Rmd way didn't work, lets try the quarto way
+  if (length(nodes) < 0) {
```

Comment:
```suggestion
  if (length(nodes) < 1) {
```

The "Quarto way" `length(nodes)` is 0, so we should change the 0 to a 1.